### PR TITLE
Add MtM MtConnect04S development board support

### PR DESCRIPTION
**Pull request by shenmengkai**

### CODE REVIEW

**Comment by shenmengkai** [comment]
I've accepted the Contributor Agreement on Sat 03 Jun 2017.

File: projects.yaml [hunk 1]
```diff
@@ -277,4 +277,8 @@ projects:
     sam3u2c_ncs36510rf_if:
         - *module_if
         - *module_hic_sam3u2c
-        - records/board/ncs36510rf.yaml        
+        - records/board/ncs36510rf.yaml
+    lpc11u35_mtconnect04s:
```
Comment (c1728p9) [inline projects.yaml]: Interface firmware needs to end with `_if`, so this should be `lpc11u35_mtconnect04s_if`. Also, you'll want to add your target to `SUPPORTED_CONFIGURATIONS` and `PROJECT_RELEASE_INFO` in test/info.py so it is recognized by testing and created as part of the release process.

**Comment by shenmengkai** [comment]
@c1728p9 I've updated the model name and the info.py, I did not find the definition of PROJECT_RELEASE_INFO and SUPPORTED_CONFIGURATIONS. Our board uses lpc11u35 as programming adapter and the main MCU is nrf51822. Could you help to verify if my information is correct please.

File: test/info.py [hunk 2]
```diff
@@ -140,6 +141,7 @@
     (   0x2202,     'lpc11u35_wizwiki_w7500_eco_if',            None,               'WIZWIKI_W7500ECO'                      ),
     (   0x2203,     'lpc11u35_wizwiki_w7500p_if',               None,               'WIZWIKI_W7500P'                        ),
     (   0x1237,     'sam3u2c_ublox_evk_nina_b1_if',            'sam3u2c_bl',        'U-BLOX-EVK-NINA-B1'                    ),
+    (   0xC005,     'lpc11u35_mtconnect04s_if',                 None,               'MtM-MtConnect04S'                      ),
```
Comment (c1728p9) [inline test/info.py]: The target name should match the platform page. In this case it should be MtConnect04S corresponding to the page https://developer.mbed.org/platforms/MtConnect04S/

Comment (shenmengkai) [inline test/info.py]: @c1728p9 the URL is updated.

Comment (c1728p9) [inline test/info.py]: Hi @shenmengkai, you don't need to change the URL. It's the target name that needs to be changed `'MtM-MtConnect04S'` -> `'MtConnect04S '` on line 144

Comment (shenmengkai) [inline test/info.py]: pardon @c1728p9, I updated the target name and revert the URL. Thank you.

**Comment by shenmengkai** [comment]
@c1728p9 when it will be merged to base?

**Comment by c1728p9** [comment]
Hi @shenmengkai, can you rebase this on master. It has merge conflicts.

**Comment by shenmengkai** [comment]
@c1728p9 done.

**Comment by studavekar** [comment]
DAPLINK test looks like it failed : http://aus-raas-linux-00:8080/job/daplink_test_matrix/42/

**Comment by shenmengkai** [comment]
@c1728p9 @studavekar what is the failure? The link is not reachable. how do I proceed?

**Comment by cederom** [comment]
Hey @c1728p9 looks like skipped mac specific test failed for `k20dx_frdmk64f_if`.. All Linux and Windows tests passed..

```
12:10:33 SubTest: test_fs_contents
12:10:33 File /Volumes/MICROBIT/MICROBIT.HTM valid
12:10:33 File /Volumes/MICROBIT/DETAILS.TXT valid
12:10:33 Skipping non file item /Volumes/MICROBIT/.fseventsd
12:10:33 SubTest: test_details_txt
12:10:33 SubTest: parse_kvp_file
12:10:33 SubTest: test_details_txt
12:10:33 SubTest: parse_kvp_file
12:10:33 <test_info.TestInfo object at 0x10f242850>
12:10:36 Data matches
12:10:36 SubTest: Load good file to restore state
12:10:36 Skipping USB test on this platform
12:10:36 
12:10:36 Test: k20dx_frdmk64f_if FRDM-K64F: Failure
12:10:36   Test: daplink_test: Failure
12:10:36     Test: bin interface filetype test: Failure
12:10:36       Test: Extra Files: Failure
12:10:36         Failure: Data does not match
```

**Comment by shenmengkai** [comment]
Hi @c1728p9 @cederom @studavekar, I wonder if it is ready to merge? or anything needs me to improve?

**Comment by studavekar** [comment]
@shenmengkai only thing pending is a green flag from CI, test on MAC were failing we will re-trigger.

**Comment by cederom** [comment]
@studavekar @c1728p9 please update CI host to aus-raas-linux-00.austin.arm.com :-)

**Comment by studavekar** [comment]
looks good to merge we have all green

**Comment by shenmengkai** [comment]
Thank @c1728p9 @studavekar and @cederom, for all your help.

**Pull request merged**